### PR TITLE
Add support for `RelatedManager.add()` `through_defaults` (Django 2.2)

### PR DESCRIPTION
**Pull request by soulshake**
This is a proof of concept intended to address https://github.com/jazzband/django-taggit/issues/629.

Problems:

- I don't think this change would be compatible with Django < 2.2, at least not in its current state.
- I am not sure how to test this properly (I made an attempt but haven't gotten it working).

I would be happy to put some more work into this if it's a desired change. If so, I would appreciate some guidance on how to go about testing it.

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -21,6 +21,9 @@ playing around with the API.
             []
             >>> apple.tags.add("red", "green", "fruit")
 
+        Use the `through_defaults` argument to specify values for your custom
+        `through` model, if needed.
```
Comment (jdufresne) [inline docs/api.rst]:
```suggestion
        ``through`` model, if needed.
```

File: docs/api.rst [hunk 2]
```diff
@@ -12,7 +12,7 @@ playing around with the API.
         information.
     :param blank: Controls whether this field is required.
 
-    .. method:: add(*tags)
+    .. method:: add(*tags, through_defaults={})
```
Comment (jdufresne) [inline docs/api.rst]:
```suggestion
    .. method:: add(*tags, through_defaults=None)
```

**Comment by jdufresne** [comment]
Hi @soulshake Thanks for the PR. Are you interested in cleaning up the failing tests?

**Comment by soulshake** [comment]
@jdufresne I would be, but I wasn't sure how to go about it.

**Comment by PeteAndersen** [comment]
I did some digging - each of the test case classes use specific models to test. The tests for using custom through models all use the `OfficialFood` model, and the unit test for testing `through_defaults` should be in the `TaggableManagerOfficialTestCase` test class.

**Comment by PeteAndersen** [comment]
Testing with Django 1.11 seems to work fine, since the `through_defaults` are passed into `get_or_create()` as the `defaults` kwarg, which has support across all the targeted Django versions.

**Comment by PeteAndersen** [comment]
It would be good to change the function signature for `add()` so that through_defaults is pulled out of keyword args. With it defined as a positional argument you would have to explicitly pass in None when using any kwargs, such as in #653

**Comment by auvipy** [comment]
please fix the merge conflict

File: taggit/managers.py [hunk 3]
```diff
@@ -143,7 +143,7 @@ def _lookup_kwargs(self):
         return self.through.lookup_kwargs(self.instance)
 
     @require_instance_manager
-    def add(self, *tags, **kwargs):
+    def add(self, through_defaults=None *tags, **kwargs):
```
Comment (auvipy) [inline taggit/managers.py:146]: missing comma!

**Pull request closed without merging**